src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -7134,14 +7136,179 @@ std::pair<MergeTreeData::MutableDataPartPtr, scope_guard> MergeTreeData::cloneAn
     if (params.copy_instead_of_hardlink)
         with_copy = " (copying data)";
 
-    auto dst_part_storage = src_part_storage->freeze(
-        relative_data_path,
-        tmp_dst_part_name,
-        read_settings,
-        write_settings,
-        /* save_metadata_callback= */ {},
-        params);
+    std::shared_ptr<IDataPartStorage> dst_part_storage{};
+    try
+    {
+        dst_part_storage = src_part_storage->freeze(
+            relative_data_path,
+            tmp_dst_part_name,
+            read_settings,
+            write_settings,
+            /* save_metadata_callback= */ {},
+            params);
+    }
+    catch (...)
```

Comment:
If it is on the same disk, why even try?

Reply:
"We can also do the same if it is done within the same disk, but the disk does not support hard links." https://github.com/ClickHouse/ClickHouse/issues/59377

You made such a request in the issue description. I understand that if freeze fails, the attach operation can be performed through copy. So I used try-catch here.